--- a/src/llvm-project/compiler-rt/lib/sanitizer_common/sanitizer_platform.h
+++ b/src/llvm-project/compiler-rt/lib/sanitizer_common/sanitizer_platform.h
@@ -123,6 +123,12 @@
 #  define SANITIZER_ANDROID 0
 #endif
 
+#if defined(__OHOS__)
+#  define SANITIZER_OHOS 1
+#else
+#  define SANITIZER_OHOS 0
+#endif
+
 #if defined(__Fuchsia__)
 #  define SANITIZER_FUCHSIA 1
 #else
--- a/src/llvm-project/compiler-rt/lib/asan/asan_allocator.h
+++ b/src/llvm-project/compiler-rt/lib/asan/asan_allocator.h
@@ -199,7 +199,7 @@
 #    if defined(__powerpc64__)
 const uptr kAllocatorSize  =  0x20000000000ULL;  // 2T.
 typedef DefaultSizeClassMap SizeClassMap;
-#    elif defined(__aarch64__) && SANITIZER_ANDROID
+#    elif defined(__aarch64__) && (SANITIZER_ANDROID || SANITIZER_OHOS)
 // Android needs to support 39, 42 and 48 bit VMA.
 const uptr kAllocatorSize  =  0x2000000000ULL;  // 128G.
 typedef VeryCompactSizeClassMap SizeClassMap;
--- a/src/llvm-project/compiler-rt/lib/hwasan/hwasan_allocator.h
+++ b/src/llvm-project/compiler-rt/lib/hwasan/hwasan_allocator.h
@@ -74,7 +74,7 @@
 #if defined(HWASAN_ALIASING_MODE)
   static const uptr kSpaceSize = 1ULL << kAddressTagShift;
   typedef __sanitizer::DefaultSizeClassMap SizeClassMap;
-#elif SANITIZER_LINUX && !SANITIZER_ANDROID
+#elif SANITIZER_LINUX && !SANITIZER_ANDROID && !SANITIZER_OHOS
   static const uptr kSpaceSize = 0x40000000000ULL;  // 4T.
   typedef __sanitizer::DefaultSizeClassMap SizeClassMap;
 #else
--- a/src/llvm-project/compiler-rt/lib/sanitizer_common/sanitizer_platform_limits_posix.cpp
+++ b/src/llvm-project/compiler-rt/lib/sanitizer_common/sanitizer_platform_limits_posix.cpp
@@ -19,6 +19,11 @@
 // To get this "true" dirent definition, we undefine _FILE_OFFSET_BITS below.
 #undef _FILE_OFFSET_BITS
 #undef _TIME_BITS
+// OHOS: include sys/sysinfo.h before any header that defines _LINUX_KERNEL_H,
+// otherwise bionic's sys/sysinfo.h skips struct sysinfo definition.
+#if defined(__OHOS__)
+#include <sys/sysinfo.h>
+#endif
 #endif
 
 // Must go after undef _FILE_OFFSET_BITS.
