diff --git a/library/std/src/num/f128.rs b/library/std/src/num/f128.rs
--- a/library/std/src/num/f128.rs
+++ b/library/std/src/num/f128.rs
@@ -29,7 +29,7 @@
     /// ```
     /// #![feature(f128)]
     /// # #[cfg(not(miri))]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let x = 2.0_f128;
     /// let abs_difference = (x.powf(2.0) - (x * x)).abs();
@@ -59,7 +59,7 @@
     /// ```
     /// #![feature(f128)]
     /// # #[cfg(not(miri))]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let one = 1.0f128;
     /// // e^1
@@ -91,7 +91,7 @@
     /// ```
     /// #![feature(f128)]
     /// # #[cfg(not(miri))]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let f = 2.0f128;
     ///
@@ -123,7 +123,7 @@
     /// ```
     /// #![feature(f128)]
     /// # #[cfg(not(miri))]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let one = 1.0f128;
     /// // e^1
@@ -140,7 +140,7 @@
     /// ```
     /// #![feature(f128)]
     /// # #[cfg(not(miri))]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// assert_eq!(0_f128.ln(), f128::NEG_INFINITY);
     /// assert!((-42_f128).ln().is_nan());
@@ -172,7 +172,7 @@
     /// ```
     /// #![feature(f128)]
     /// # #[cfg(not(miri))]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let five = 5.0f128;
     ///
@@ -187,7 +187,7 @@
     /// ```
     /// #![feature(f128)]
     /// # #[cfg(not(miri))]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// assert_eq!(0_f128.log(10.0), f128::NEG_INFINITY);
     /// assert!((-42_f128).log(10.0).is_nan());
@@ -215,7 +215,7 @@
     /// ```
     /// #![feature(f128)]
     /// # #[cfg(not(miri))]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let two = 2.0f128;
     ///
@@ -230,7 +230,7 @@
     /// ```
     /// #![feature(f128)]
     /// # #[cfg(not(miri))]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// assert_eq!(0_f128.log2(), f128::NEG_INFINITY);
     /// assert!((-42_f128).log2().is_nan());
@@ -258,7 +258,7 @@
     /// ```
     /// #![feature(f128)]
     /// # #[cfg(not(miri))]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let ten = 10.0f128;
     ///
@@ -273,7 +273,7 @@
     /// ```
     /// #![feature(f128)]
     /// # #[cfg(not(miri))]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// assert_eq!(0_f128.log10(), f128::NEG_INFINITY);
     /// assert!((-42_f128).log10().is_nan());
@@ -303,7 +303,7 @@
     /// ```
     /// #![feature(f128)]
     /// # #[cfg(not(miri))]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let x = 8.0f128;
     ///
@@ -340,7 +340,7 @@
     /// ```
     /// #![feature(f128)]
     /// # #[cfg(not(miri))]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let x = 2.0f128;
     /// let y = 3.0f128;
@@ -371,7 +371,7 @@
     /// ```
     /// #![feature(f128)]
     /// # #[cfg(not(miri))]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let x = std::f128::consts::FRAC_PI_2;
     ///
@@ -400,7 +400,7 @@
     /// ```
     /// #![feature(f128)]
     /// # #[cfg(not(miri))]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let x = 2.0 * std::f128::consts::PI;
     ///
@@ -432,7 +432,7 @@
     /// ```
     /// #![feature(f128)]
     /// # #[cfg(not(miri))]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let x = std::f128::consts::FRAC_PI_4;
     /// let abs_difference = (x.tan() - 1.0).abs();
@@ -501,7 +501,7 @@
     /// ```
     /// #![feature(f128)]
     /// # #[cfg(not(miri))]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let f = std::f128::consts::FRAC_PI_4;
     ///
@@ -536,7 +536,7 @@
     /// ```
     /// #![feature(f128)]
     /// # #[cfg(not(miri))]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let f = 1.0f128;
     ///
@@ -575,7 +575,7 @@
     /// ```
     /// #![feature(f128)]
     /// # #[cfg(not(miri))]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// // Positive angles measured counter-clockwise
     /// // from positive x axis
@@ -618,7 +618,7 @@
     /// ```
     /// #![feature(f128)]
     /// # #[cfg(not(miri))]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let x = std::f128::consts::FRAC_PI_4;
     /// let f = x.sin_cos();
@@ -654,7 +654,7 @@
     /// ```
     /// #![feature(f128)]
     /// # #[cfg(not(miri))]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let x = 1e-8_f128;
     ///
@@ -691,7 +691,7 @@
     /// ```
     /// #![feature(f128)]
     /// # #[cfg(not(miri))]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let x = 1e-8_f128;
     ///
@@ -737,7 +737,7 @@
     /// ```
     /// #![feature(f128)]
     /// # #[cfg(not(miri))]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let e = std::f128::consts::E;
     /// let x = 1.0f128;
@@ -773,7 +773,7 @@
     /// ```
     /// #![feature(f128)]
     /// # #[cfg(not(miri))]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let e = std::f128::consts::E;
     /// let x = 1.0f128;
@@ -809,7 +809,7 @@
     /// ```
     /// #![feature(f128)]
     /// # #[cfg(not(miri))]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let e = std::f128::consts::E;
     /// let x = 1.0f128;
@@ -910,7 +910,7 @@
     /// ```
     /// #![feature(f128)]
     /// # #[cfg(not(miri))]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let e = std::f128::consts::E;
     /// let f = e.tanh().atanh();
@@ -945,7 +945,7 @@
     /// #![feature(f128)]
     /// #![feature(float_gamma)]
     /// # #[cfg(not(miri))]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let x = 5.0f128;
     ///
@@ -981,7 +981,7 @@
     /// #![feature(f128)]
     /// #![feature(float_gamma)]
     /// # #[cfg(not(miri))]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let x = 2.0f128;
     ///
@@ -1017,7 +1017,7 @@
     /// #![feature(f128)]
     /// #![feature(float_erf)]
     /// # #[cfg(not(miri))]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     /// /// The error function relates what percent of a normal distribution lies
     /// /// within `x` standard deviations (scaled by `1/sqrt(2)`).
     /// fn within_standard_deviations(x: f128) -> f128 {
@@ -1057,7 +1057,7 @@
     /// #![feature(f128)]
     /// #![feature(float_erf)]
     /// # #[cfg(not(miri))]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     /// let x: f128 = 0.123;
     ///
     /// let one = x.erf() + x.erfc();
diff --git a/library/std/src/num/f16.rs b/library/std/src/num/f16.rs
