diff --git a/library/std/src/num/f128.rs b/library/std/src/num/f128.rs
--- a/library/std/src/num/f128.rs
+++ b/library/std/src/num/f128.rs
@@ -34,7 +34,7 @@
     ///
     /// ```
     /// #![feature(f128)]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let x = 2.0_f128;
     /// let abs_difference = (x.powf(2.0) - (x * x)).abs();
@@ -64,7 +64,7 @@
     ///
     /// ```
     /// #![feature(f128)]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let one = 1.0f128;
     /// // e^1
@@ -95,7 +95,7 @@
     ///
     /// ```
     /// #![feature(f128)]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let f = 2.0f128;
     ///
@@ -126,7 +126,7 @@
     ///
     /// ```
     /// #![feature(f128)]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let one = 1.0f128;
     /// // e^1
@@ -142,7 +142,7 @@
     /// Non-positive values:
     /// ```
     /// #![feature(f128)]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// assert_eq!(0_f128.ln(), f128::NEG_INFINITY);
     /// assert!((-42_f128).ln().is_nan());
@@ -173,7 +173,7 @@
     ///
     /// ```
     /// #![feature(f128)]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let five = 5.0f128;
     ///
@@ -187,7 +187,7 @@
     /// Non-positive values:
     /// ```
     /// #![feature(f128)]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// assert_eq!(0_f128.log(10.0), f128::NEG_INFINITY);
     /// assert!((-42_f128).log(10.0).is_nan());
@@ -214,7 +214,7 @@
     ///
     /// ```
     /// #![feature(f128)]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let two = 2.0f128;
     ///
@@ -228,7 +228,7 @@
     /// Non-positive values:
     /// ```
     /// #![feature(f128)]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// assert_eq!(0_f128.log2(), f128::NEG_INFINITY);
     /// assert!((-42_f128).log2().is_nan());
@@ -255,7 +255,7 @@
     ///
     /// ```
     /// #![feature(f128)]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let ten = 10.0f128;
     ///
@@ -269,7 +269,7 @@
     /// Non-positive values:
     /// ```
     /// #![feature(f128)]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// assert_eq!(0_f128.log10(), f128::NEG_INFINITY);
     /// assert!((-42_f128).log10().is_nan());
@@ -298,7 +298,7 @@
     ///
     /// ```
     /// #![feature(f128)]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let x = 8.0f128;
     ///
@@ -334,7 +334,7 @@
     ///
     /// ```
     /// #![feature(f128)]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let x = 2.0f128;
     /// let y = 3.0f128;
@@ -364,7 +364,7 @@
     ///
     /// ```
     /// #![feature(f128)]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let x = std::f128::consts::FRAC_PI_2;
     ///
@@ -392,7 +392,7 @@
     ///
     /// ```
     /// #![feature(f128)]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let x = 2.0 * std::f128::consts::PI;
     ///
@@ -423,7 +423,7 @@
     ///
     /// ```
     /// #![feature(f128)]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let x = std::f128::consts::FRAC_PI_4;
     /// let abs_difference = (x.tan() - 1.0).abs();
@@ -455,7 +455,7 @@
     ///
     /// ```
     /// #![feature(f128)]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let f = std::f128::consts::FRAC_PI_4;
     ///
@@ -490,7 +490,7 @@
     ///
     /// ```
     /// #![feature(f128)]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let f = std::f128::consts::FRAC_PI_4;
     ///
@@ -524,7 +524,7 @@
     ///
     /// ```
     /// #![feature(f128)]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let f = 1.0f128;
     ///
@@ -564,7 +564,7 @@
     ///
     /// ```
     /// #![feature(f128)]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// // Positive angles measured counter-clockwise
     /// // from positive x axis
@@ -606,7 +606,7 @@
     ///
     /// ```
     /// #![feature(f128)]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let x = std::f128::consts::FRAC_PI_4;
     /// let f = x.sin_cos();
@@ -642,7 +642,7 @@
     ///
     /// ```
     /// #![feature(f128)]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let x = 1e-8_f128;
     ///
@@ -678,7 +678,7 @@
     ///
     /// ```
     /// #![feature(f128)]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let x = 1e-8_f128;
     ///
@@ -693,7 +693,7 @@
     /// Out-of-range values:
     /// ```
     /// #![feature(f128)]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// assert_eq!((-1.0_f128).ln_1p(), f128::NEG_INFINITY);
     /// assert!((-2.0_f128).ln_1p().is_nan());
@@ -722,7 +722,7 @@
     ///
     /// ```
     /// #![feature(f128)]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let e = std::f128::consts::E;
     /// let x = 1.0f128;
@@ -757,7 +757,7 @@
     ///
     /// ```
     /// #![feature(f128)]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let e = std::f128::consts::E;
     /// let x = 1.0f128;
@@ -792,7 +792,7 @@
     ///
     /// ```
     /// #![feature(f128)]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let e = std::f128::consts::E;
     /// let x = 1.0f128;
@@ -824,7 +824,7 @@
     ///
     /// ```
     /// #![feature(f128)]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let x = 1.0f128;
     /// let f = x.sinh().asinh();
@@ -854,7 +854,7 @@
     ///
     /// ```
     /// #![feature(f128)]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let x = 1.0f128;
     /// let f = x.cosh().acosh();
@@ -884,7 +884,7 @@
     ///
     /// ```
     /// #![feature(f128)]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let x = std::f128::consts::FRAC_PI_6;
     /// let f = x.tanh().atanh();
@@ -918,7 +918,7 @@
     /// ```
     /// #![feature(f128)]
     /// #![feature(float_gamma)]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let x = 5.0f128;
     ///
@@ -953,7 +953,7 @@
     /// ```
     /// #![feature(f128)]
     /// #![feature(float_gamma)]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     ///
     /// let x = 2.0f128;
     ///
@@ -988,7 +988,7 @@
     /// ```
     /// #![feature(f128)]
     /// #![feature(float_erf)]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     /// /// The error function relates what percent of a normal distribution lies
     /// /// within `x` standard deviations (scaled by `1/sqrt(2)`).
     /// fn within_standard_deviations(x: f128) -> f128 {
@@ -1027,7 +1027,7 @@
     /// ```
     /// #![feature(f128)]
     /// #![feature(float_erf)]
-    /// # #[cfg(target_has_reliable_f128_math)] {
+    /// # #[cfg(all(target_has_reliable_f128_math, not(target_env = "ohos")))] {
     /// let x: f128 = 0.123;
     ///
     /// let one = x.erf() + x.erfc();
diff --git a/library/std/src/num/f16.rs b/library/std/src/num/f16.rs
--- a/library/std/src/num/f16.rs
+++ b/library/std/src/num/f16.rs
@@ -34,7 +34,7 @@
     ///
     /// ```
     /// #![feature(f16)]
-    /// # #[cfg(target_has_reliable_f16_math)] {
+    /// # #[cfg(all(target_has_reliable_f16_math, not(target_env = "ohos")))] {
     ///
     /// let x = 2.0_f16;
     /// let abs_difference = (x.powf(2.0) - (x * x)).abs();
@@ -64,7 +64,7 @@
     ///
     /// ```
     /// #![feature(f16)]
-    /// # #[cfg(target_has_reliable_f16_math)] {
+    /// # #[cfg(all(target_has_reliable_f16_math, not(target_env = "ohos")))] {
     ///
     /// let one = 1.0f16;
     /// // e^1
@@ -95,7 +95,7 @@
     ///
     /// ```
     /// #![feature(f16)]
-    /// # #[cfg(target_has_reliable_f16_math)] {
+    /// # #[cfg(all(target_has_reliable_f16_math, not(target_env = "ohos")))] {
     ///
     /// let f = 2.0f16;
     ///
@@ -126,7 +126,7 @@
     ///
     /// ```
     /// #![feature(f16)]
-    /// # #[cfg(target_has_reliable_f16_math)] {
+    /// # #[cfg(all(target_has_reliable_f16_math, not(target_env = "ohos")))] {
     ///
     /// let one = 1.0f16;
     /// // e^1
@@ -142,7 +142,7 @@
     /// Non-positive values:
     /// ```
     /// #![feature(f16)]
-    /// # #[cfg(target_has_reliable_f16_math)] {
+    /// # #[cfg(all(target_has_reliable_f16_math, not(target_env = "ohos")))] {
     ///
     /// assert_eq!(0_f16.ln(), f16::NEG_INFINITY);
     /// assert!((-42_f16).ln().is_nan());
@@ -173,7 +173,7 @@
     ///
     /// ```
     /// #![feature(f16)]
-    /// # #[cfg(target_has_reliable_f16_math)] {
+    /// # #[cfg(all(target_has_reliable_f16_math, not(target_env = "ohos")))] {
     ///
     /// let five = 5.0f16;
     ///
@@ -187,7 +187,7 @@
     /// Non-positive values:
     /// ```
     /// #![feature(f16)]
-    /// # #[cfg(target_has_reliable_f16_math)] {
+    /// # #[cfg(all(target_has_reliable_f16_math, not(target_env = "ohos")))] {
     ///
     /// assert_eq!(0_f16.log(10.0), f16::NEG_INFINITY);
     /// assert!((-42_f16).log(10.0).is_nan());
@@ -214,7 +214,7 @@
     ///
     /// ```
     /// #![feature(f16)]
-    /// # #[cfg(target_has_reliable_f16_math)] {
+    /// # #[cfg(all(target_has_reliable_f16_math, not(target_env = "ohos")))] {
     ///
     /// let two = 2.0f16;
     ///
@@ -228,7 +228,7 @@
     /// Non-positive values:
     /// ```
     /// #![feature(f16)]
-    /// # #[cfg(target_has_reliable_f16_math)] {
+    /// # #[cfg(all(target_has_reliable_f16_math, not(target_env = "ohos")))] {
     ///
     /// assert_eq!(0_f16.log2(), f16::NEG_INFINITY);
     /// assert!((-42_f16).log2().is_nan());
@@ -255,7 +255,7 @@
     ///
     /// ```
     /// #![feature(f16)]
-    /// # #[cfg(target_has_reliable_f16_math)] {
+    /// # #[cfg(all(target_has_reliable_f16_math, not(target_env = "ohos")))] {
     ///
     /// let ten = 10.0f16;
     ///
@@ -269,7 +269,7 @@
     /// Non-positive values:
     /// ```
     /// #![feature(f16)]
-    /// # #[cfg(target_has_reliable_f16_math)] {
+    /// # #[cfg(all(target_has_reliable_f16_math, not(target_env = "ohos")))] {
     ///
     /// assert_eq!(0_f16.log10(), f16::NEG_INFINITY);
     /// assert!((-42_f16).log10().is_nan());
@@ -300,7 +300,7 @@
     ///
     /// ```
     /// #![feature(f16)]
-    /// # #[cfg(target_has_reliable_f16_math)] {
+    /// # #[cfg(all(target_has_reliable_f16_math, not(target_env = "ohos")))] {
     ///
     /// let x = 2.0f16;
     /// let y = 3.0f16;
@@ -330,7 +330,7 @@
     ///
     /// ```
     /// #![feature(f16)]
-    /// # #[cfg(target_has_reliable_f16_math)] {
+    /// # #[cfg(all(target_has_reliable_f16_math, not(target_env = "ohos")))] {
     ///
     /// let x = std::f16::consts::FRAC_PI_2;
     ///
@@ -358,7 +358,7 @@
     ///
     /// ```
     /// #![feature(f16)]
-    /// # #[cfg(target_has_reliable_f16_math)] {
+    /// # #[cfg(all(target_has_reliable_f16_math, not(target_env = "ohos")))] {
     ///
     /// let x = 2.0 * std::f16::consts::PI;
     ///
@@ -389,7 +389,7 @@
     ///
     /// ```
     /// #![feature(f16)]
-    /// # #[cfg(target_has_reliable_f16_math)] {
+    /// # #[cfg(all(target_has_reliable_f16_math, not(target_env = "ohos")))] {
     ///
     /// let x = std::f16::consts::FRAC_PI_4;
     /// let abs_difference = (x.tan() - 1.0).abs();
@@ -421,7 +421,7 @@
     ///
     /// ```
     /// #![feature(f16)]
-    /// # #[cfg(target_has_reliable_f16_math)] {
+    /// # #[cfg(all(target_has_reliable_f16_math, not(target_env = "ohos")))] {
     ///
     /// let f = std::f16::consts::FRAC_PI_4;
     ///
@@ -456,7 +456,7 @@
     ///
     /// ```
     /// #![feature(f16)]
-    /// # #[cfg(target_has_reliable_f16_math)] {
+    /// # #[cfg(all(target_has_reliable_f16_math, not(target_env = "ohos")))] {
     ///
     /// let f = std::f16::consts::FRAC_PI_4;
     ///
@@ -490,7 +490,7 @@
     ///
     /// ```
     /// #![feature(f16)]
-    /// # #[cfg(target_has_reliable_f16_math)] {
+    /// # #[cfg(all(target_has_reliable_f16_math, not(target_env = "ohos")))] {
     ///
     /// let f = 1.0f16;
     ///
@@ -530,7 +530,7 @@
     ///
     /// ```
     /// #![feature(f16)]
-    /// # #[cfg(target_has_reliable_f16_math)] {
+    /// # #[cfg(all(target_has_reliable_f16_math, not(target_env = "ohos")))] {
     ///
     /// // Positive angles measured counter-clockwise
     /// // from positive x axis
@@ -572,7 +572,7 @@
     ///
     /// ```
     /// #![feature(f16)]
-    /// # #[cfg(target_has_reliable_f16_math)] {
+    /// # #[cfg(all(target_has_reliable_f16_math, not(target_env = "ohos")))] {
     ///
     /// let x = std::f16::consts::FRAC_PI_4;
     /// let f = x.sin_cos();
@@ -608,7 +608,7 @@
     ///
     /// ```
     /// #![feature(f16)]
-    /// # #[cfg(target_has_reliable_f16_math)] {
+    /// # #[cfg(all(target_has_reliable_f16_math, not(target_env = "ohos")))] {
     ///
     /// let x = 1e-4_f16;
     ///
@@ -644,7 +644,7 @@
     ///
     /// ```
     /// #![feature(f16)]
-    /// # #[cfg(target_has_reliable_f16_math)] {
+    /// # #[cfg(all(target_has_reliable_f16_math, not(target_env = "ohos")))] {
     ///
     /// let x = 1e-4_f16;
     ///
@@ -659,7 +659,7 @@
     /// Out-of-range values:
     /// ```
     /// #![feature(f16)]
-    /// # #[cfg(target_has_reliable_f16_math)] {
+    /// # #[cfg(all(target_has_reliable_f16_math, not(target_env = "ohos")))] {
     ///
     /// assert_eq!((-1.0_f16).ln_1p(), f16::NEG_INFINITY);
     /// assert!((-2.0_f16).ln_1p().is_nan());
@@ -688,7 +688,7 @@
     ///
     /// ```
     /// #![feature(f16)]
-    /// # #[cfg(target_has_reliable_f16_math)] {
+    /// # #[cfg(all(target_has_reliable_f16_math, not(target_env = "ohos")))] {
     ///
     /// let e = std::f16::consts::E;
     /// let x = 1.0f16;
@@ -723,7 +723,7 @@
     ///
     /// ```
     /// #![feature(f16)]
-    /// # #[cfg(target_has_reliable_f16_math)] {
+    /// # #[cfg(all(target_has_reliable_f16_math, not(target_env = "ohos")))] {
     ///
     /// let e = std::f16::consts::E;
     /// let x = 1.0f16;
@@ -758,7 +758,7 @@
     ///
     /// ```
     /// #![feature(f16)]
-    /// # #[cfg(target_has_reliable_f16_math)] {
+    /// # #[cfg(all(target_has_reliable_f16_math, not(target_env = "ohos")))] {
     ///
     /// let e = std::f16::consts::E;
     /// let x = 1.0f16;
@@ -790,7 +790,7 @@
     ///
     /// ```
     /// #![feature(f16)]
-    /// # #[cfg(target_has_reliable_f16_math)] {
+    /// # #[cfg(all(target_has_reliable_f16_math, not(target_env = "ohos")))] {
     ///
     /// let x = 1.0f16;
     /// let f = x.sinh().asinh();
@@ -820,7 +820,7 @@
     ///
     /// ```
     /// #![feature(f16)]
-    /// # #[cfg(target_has_reliable_f16_math)] {
+    /// # #[cfg(all(target_has_reliable_f16_math, not(target_env = "ohos")))] {
     ///
     /// let x = 1.0f16;
     /// let f = x.cosh().acosh();
@@ -850,7 +850,7 @@
     ///
     /// ```
     /// #![feature(f16)]
-    /// # #[cfg(target_has_reliable_f16_math)] {
+    /// # #[cfg(all(target_has_reliable_f16_math, not(target_env = "ohos")))] {
     ///
     /// let x = std::f16::consts::FRAC_PI_6;
     /// let f = x.tanh().atanh();
@@ -883,7 +883,7 @@
     ///
     /// ```
     /// #![feature(f16)]
-    /// # #[cfg(target_has_reliable_f16_math)] {
+    /// # #[cfg(all(target_has_reliable_f16_math, not(target_env = "ohos")))] {
     ///
     /// let x = 5.0f16;
     ///
@@ -917,7 +917,7 @@
     ///
     /// ```
     /// #![feature(f16)]
-    /// # #[cfg(target_has_reliable_f16_math)] {
+    /// # #[cfg(all(target_has_reliable_f16_math, not(target_env = "ohos")))] {
     ///
     /// let x = 2.0f16;
     ///
@@ -951,7 +951,7 @@
     ///
     /// ```
     /// #![feature(f16)]
-    /// # #[cfg(target_has_reliable_f16_math)] {
+    /// # #[cfg(all(target_has_reliable_f16_math, not(target_env = "ohos")))] {
     /// /// The error function relates what percent of a normal distribution lies
     /// /// within `x` standard deviations (scaled by `1/sqrt(2)`).
     /// fn within_standard_deviations(x: f16) -> f16 {
@@ -989,7 +989,7 @@
     ///
     /// ```
     /// #![feature(f16)]
-    /// # #[cfg(target_has_reliable_f16_math)] {
+    /// # #[cfg(all(target_has_reliable_f16_math, not(target_env = "ohos")))] {
     /// let x: f16 = 0.123;
     ///
     /// let one = x.erf() + x.erfc();
